src/runtime_src/core/pcie/driver/aws/kernel/mgmt/mgmt-bit.c

```diff
@@ -418,17 +429,20 @@ long load_boot_firmware(struct awsmgmt_dev *lro)
 	printk(KERN_DEBUG "%s:%s:%d\n", __func__, __FILE__, __LINE__);
 	memset(&bit_header, 0, sizeof(bit_header));
 
-	snprintf(fw_name, sizeof(fw_name), "xilinx/%04x-%04x-%04x-%016llx.dsabin",
-		 le16_to_cpu(lro->user_pci_dev->vendor),
-		 le16_to_cpu(lro->user_pci_dev->device),
-		 le16_to_cpu(lro->user_pci_dev->subsystem_device),
-		 le64_to_cpu(lro->feature_id));
-
+	//try xsabin suffix first, if failed, then try dsabin suffix
+	create_fw_name(fw_name, sizeof(fw_name), lro, "xsabin");
 	err = request_firmware(&fw, fw_name, &lro->pci_dev->dev);
 
 	if (err) {
 		printk(KERN_WARNING "Unable to find firmware %s\n", fw_name);
-		return err;
+
+		create_fw_name(fw_name, sizeof(fw_name), lro, "dsabin");
+		printk(KERN_INFO "Try to find firmware %s\n", fw_name);
```

Comment:
Could the message be updated to be more "user" friendly indicating that the legacy fw image is being sought after.  For example:
`printk(KERN_WARNING "Searching for legacy firmware image %s\n", fw_name);`

This might be important for legacy platforms will always produce a the warning on line 437 stating that the firmware image could not be found.

Reply:
Thanks! Added legacy info into the warning message. Please see the most recent commit.